<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>React expects a unique identifier for performance optimizations. Using generated values like <code>Math.random()</code> or <code>Date.now()</code> is discouraged as it would force a re-render every time and might also provoke bugs if values collide.</p>
</div>
<div class="paragraph">
<p>Instead, use a unique attribute like an ID or a combination of attributes.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function generateButtons(props) {
  return props.buttons.map((button, index) =&gt; {
    &lt;Button key={Math.random()}&gt;{button.number}&lt;/Button&gt;
  });
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function generateButtons(props) {
  return props.buttons.map((button, index) =&gt; {
    &lt;Button key={button.number}&gt;{button.number}&lt;/Button&gt;
  });
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://reactjs.org/docs/reconciliation.html#recursing-on-children">Recursing On Children</a> - React API reference</p>
</li>
<li>
<p><a data-rspec-id="S6477" class="rspec-auto-link">S6477</a></p>
</li>
<li>
<p><a data-rspec-id="S6479" class="rspec-auto-link">S6479</a></p>
</li>
</ul>
</div>
</div>
</div>